packages/dd-trace/src/opentracing/span.js

```diff
@@ -262,6 +263,10 @@ class DatadogSpan {
           .padStart(8, '0')
           .padEnd(16, '0')
       }
+
+      if (fields.apmTracingEnabled === false) {
+        spanContext._trace.tags[APM_TRACING_ENABLED_KEY] = 0
```

Comment:
why don't we just check for the singleton here too ? instead of having a whole different variable ?

Reply:
al the moment, apm tracing is disabled when appsec standalone is true but in the future there might be other products with the need to disable apm tracing, thats why it is a different variable.
And also if we check here the `standalone` singleton, opentracing span would have a dependency with appsec...

Reply:
naming is horrible tho, the tracing is obviously enabled, it's just changing a tag.
Also I'm not sure how useful it is to futur-proof this right now when we're still in a very experimental phase. Feels a tiny bit over-engineered no ?
And the singleton isn't really appsec, you could literally put it in config.js and it would still work the exact same. The Appsec part is just from an arbitrary path you choose

Reply:
I think they are two different things:
- standalone ASM is configured by `DD_EXPERIMENTAL_APPSEC_STANDALONE_ENABLED`. It "disables" apm tracing, adds a tag when appsec events occurs and also interferes in the priority sampler to change traces priority based on span tags (appsec specific tags). So these requirements are specific to appsec, they are not generic and that's why `appsec/standalone` singleton is in the appsec namespace.

- Other aspect is to disable apm tracing. I know, we only include a tag in the trace root span to indicate or pretend is disabled (but that's what RFC says). So I see this as something generic and that's why I created `apmTracingEnabled` config property to isolate the span context creation with appsec configuration.

I should have included an explanation like this in the PR description.

Reply:
- ![image](https://github.com/DataDog/dd-trace-js/assets/70517128/a560eb74-dc7b-43c9-a0a1-3d5e796b0a05)
- ![image](https://github.com/DataDog/dd-trace-js/assets/70517128/c2a5a058-0676-4fd3-93fa-d32a653c6a34)

you say the same thing twice xD

Reply:
you're trying to tell me they're two different thing by telling me they are the same thing ? :thinking:

Reply:
but they are not the same. One depends on the other :)

For example IAST enables log collection. And also profiling. Log collection module doesn't have a dependency neither iast nor profiling.
I see apm tracing case the same.

```javascript
const hasTelemetryLogsUsingFeatures =
      isTrue(DD_IAST_ENABLED) ||
      isTrue(DD_PROFILING_ENABLED) ||
      (typeof DD_INJECTION_ENABLED === 'string' && DD_INJECTION_ENABLED.split(',').includes('profiling'))
        ? true
        : undefined
    this._setBoolean(env, 'telemetry.logCollection', coalesce(DD_TELEMETRY_LOG_COLLECTION_ENABLED,
      hasTelemetryLogsUsingFeatures))
```

Reply:
ok, other option. appsec/standalone module could subscribe to `dd-trace:span:start`, check if is a root span and add the tag. 
That way I could rid of `config.apmTracingEnabled` and the tracer and span modifications.

better?

Reply:
even better, that makes the whole feature in one line no ?
![image](https://github.com/DataDog/dd-trace-js/assets/70517128/c5c4d178-64a5-43d1-88a6-426d0bb35e69)

Reply:
if it is approved by apm i'd have no problem.

Reply:
Hi, want to share my opinion here:

- Other properties in the span coming from the config, are made by `fields` thing, I think that we shouldn't call to other method into the Span class, we should do the same that the tracer is already doing for other properties like `traceId128BitGenerationEnabled`

- About this:
> even better, that makes the whole feature in one line no ?

Adding the tag in the tracer, out of the `Span` could be a good solution

- About using `this._config.appsec.standalone.enabled` or `this._config.apmTracingEnabled`... yes, could be good if we can prevent creating `apmTracingEnabled` field for now, we could create it in the future if more features need to work in the same way.

Reply:
let's see what APM thinks about this

Reply:
actually, the change would be in `tracer.js` as pointed out by ugaitz to continue using `fields` argument.

`this._apmTracingEnabled = config.appsec.standalone.enabled`

Reply:
And `apmTracingEnabled` name is derived from the env var proposed at the beginning in the RFC.
I'm open to suggestions if you don't like it

Reply:
I think overall APM should be the one to decide how they want to do this @rochdev

Reply:
or @Qard (as I believe you were talking about this in the guild)

Reply:
ok, i've just removed `apmTracingEnabled` config property but kept `DatadogTracer._apmTracingEnabled` to pass the flag as a field when creating the Span. PTAL.

Reply:
resolving this conversation because it no longer applies and could be confusing